ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingHooks.kt

```diff
@@ -0,0 +1,21 @@
+package io.ktor.server.routing
+
+import io.ktor.server.application.*
+
+/**
+ * Hook that will be triggered when a call was routed and before any other handlers.
+ */
+public object CallRouted : Hook<(RoutingApplicationCall) -> Unit> {
+    override fun install(application: ApplicationCallPipeline, handler: (RoutingApplicationCall) -> Unit) {
+        application.environment?.monitor?.subscribe(Routing.RoutingCallStarted) { call -> handler(call) }
```

Comment:
I'm not sure that monitoring is a proper way of doing that, cause it would be really handy to throw an exception here in some cases

Reply:
We have `CallFailed` for handling exceptions while call routed is just an event that happens at the same time as `RoutingCallStarted` (basically it is just a wrapper over the later that is new-api-friendly)

Reply:
I'll make it internal anyway, so that we can describe later